portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -122,21 +134,31 @@ private String convertToJsonUsingGson(ArrayList<String> lst) {
   // A simple HTTP handler to extract text input from submitted web form and respond that context back to the user.
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     
-    String title = request.getParameter("title");
-    String name = request.getParameter("name");
-    long timestamp = System.currentTimeMillis();
-
-    Entity taskEntity = new Entity("Task");
-    taskEntity.setProperty("title", title);
-    taskEntity.setProperty("name", name);
-    taskEntity.setProperty("timestamp", timestamp);
-    datastore.put(taskEntity);
-
-    response.setContentType("text/html;");
-    response.getWriter().println(title);
-    response.getWriter().println(name);
-    response.sendRedirect("/index.html");
-    
+    UserService userService = UserServiceFactory.getUserService();
+
+    // Must be logged in to post comments
+    System.out.println("User:" + userService.isUserLoggedIn());
+    if (!userService.isUserLoggedIn()) {
+      response.getWriter().println("<p>You must be logged in to submit a comment!</p>");
+      return;
```

Comment:
This is okay for this exercise. In the real world, the UI should not even allow a non-login user to make a post request.  The UI should provide hint for users to login before showing button to post comments.